code/controllers/subsystem/ticker.dm

```diff
@@ -191,36 +194,73 @@ SUBSYSTEM_DEF(ticker)
 				send_tip_of_the_round()
 				tipped = TRUE
 
+			if(timeLeft <= 300 && !pre_setup_completed)
+				//Setup gamemode maps 30 seconds before roundstart.
+				if(!pre_setup())
+					//The mode failed to pre-setup
+					//This means that either there are insufficient players to run the gamemode
+					//Or there are no runnable gamemodes.
+					//In this case, we will resort to running extended.
+					to_chat(world, "<span class='boldannounce'>Pre-setup failed; Forcing gamemode to extended. Please file a github report including the current round ID: [GLOB.round_id].</span>")
+					message_admins("Error: [mode.setup_error]")
+					mode = null
+					GLOB.master_mode = "extended"
+					//Lets try this again.
+					if(!pre_setup())
+						//This should be impossible to reach this point, unless someone edits and breaks the standard gamemode.
+						to_chat(world, "<span class='warning'>WARNING: Pre-setup failed on fallback gamemode: extended. This should never happen; something is seriously wrong.</span>")
+						send2irc("Server", "WARNING: Presetup failed! Round ID: [GLOB.round_id].")
+						//Emergency start the game.
+						//Pre_setup simply handles picking the gamemode and map setup, so we will force
+						//pick extended. (No map setup required for it.)
+						emergency_start = TRUE
+						pre_setup_completed = TRUE
+						mode = config.pick_mode("extended")
+						return
+				pre_setup_completed = TRUE
+
 			if(timeLeft <= 0)
 				current_state = GAME_STATE_SETTING_UP
 				Master.SetRunLevel(RUNLEVEL_SETUP)
 				if(start_immediately)
 					fire()
 
 		if(GAME_STATE_SETTING_UP)
-			if(!pre_setup())
+			if(fail_counter >= 2)
```

Comment:
put this at the end and possibly define magic number

Reply:
why should it be at the end?

Reply:
because that would give time for an admin to change the gamemode if setup fails